src/main/kotlin/se/uulm/snowballr/backend/repository/association/ReadingListTableRepo.kt

```diff
@@ -0,0 +1,89 @@
+package se.uulm.snowballr.backend.repository.association
+
+import org.jetbrains.exposed.sql.SqlExpressionBuilder.eq
+import org.jetbrains.exposed.sql.and
+import org.jetbrains.exposed.sql.deleteWhere
+import org.jetbrains.exposed.sql.insert
+import org.jetbrains.exposed.sql.selectAll
+import se.uulm.snowballr.backend.db.IDatabase
+import se.uulm.snowballr.backend.model.dto.Paper
+import se.uulm.snowballr.backend.table.PaperTable
+import se.uulm.snowballr.backend.table.association.ReadingListTable
+import se.uulm.snowballr.backend.table.toPaper
+import java.util.UUID
+
+/**
+ * Defines an interface for repository operations related to the [ReadingListTable].
+ *
+ * This interface provides an abstraction for handling persistence and retrieval operations for reading list entries.
+ * By using this interface, the functionality for managing reading lists remains decoupled from the specifics of the
+ * database layer, promoting flexibility and maintainability in the codebase.
+ */
+interface IReadingListTableRepo {
+    /**
+     * Add the provided paper to the reading list of the provided user.
+     */
+    suspend fun createReadingListEntry(userId: UUID, paperId: UUID)
+
+    /**
+     * Remove the provided paper from the reading list of the provided user.
+     */
+    suspend fun removeReadingListEntry(userId: UUID, paperId: UUID)
+
+    /**
+     * @returns Whether the provided paper is on the reading list of the provided user.
+     */
+    suspend fun isPaperOnReadingList(userId: UUID, paperId: UUID): Boolean
+
+    /**
+     * @returns The [List] of all [Paper]s the provided user has on their reading list.
```

Comment:
`@returns` doesn't exist in KDoc.
```suggestion
     * @return Whether the provided paper is on the reading list of the provided user.
     */
    suspend fun isPaperOnReadingList(userId: UUID, paperId: UUID): Boolean

    /**
     * @return The [List] of all [Paper]s the provided user has on their reading list.
```